pkg/controller/stack/stack_controller.go

```diff
@@ -396,25 +388,6 @@ func newReconcileStackSession(
 	}
 }
 
-// gitCloneAndCheckoutCommit clones the Git repository and checkouts the specified commit hash or branch.
-func gitCloneAndCheckoutCommit(url, hash, branch, path string) error {
```

Comment:
Is this handled elsewhere now?

Reply:
Yes - its done by the automation api so this was unused. See `auth.NewLocalWorkspace`.